server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java

```diff
@@ -182,6 +184,75 @@ public void getMessageListShouldListMessageWhenTheUserHasOnlyReadRight() throws
             .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
     }
 
+    @Test
+    public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBodyAfterTheMessageMailboxHasBeenChanged() throws Exception {
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, "mailbox");
+        MailboxId otherMailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, "otherMailbox");
+
+        ComposedMessageId message = mailboxProbe.appendMessage(alice, MailboxPath.forUser(alice, "mailbox"),
+            ClassLoader.getSystemResourceAsStream("eml/twoAttachments.eml"), new Date(), false, new Flags());
+        await();
+
+        String messageId = message.getMessageId().serialize();
+
+        given()
+            .header("Authorization", aliceAccessToken.serialize())
+            .body("[[\"setMessages\", {\"update\":{\"" + messageId + "\":{\"mailboxIds\": [\"" + otherMailboxId.serialize() + "\"]}}}, \"#0\"]]")
+            .when()
+            .post("/jmap")
+            .then()
+            .statusCode(200)
+            .log();
+        await();
+
+        given()
+            .header("Authorization", aliceAccessToken.serialize())
+            .body(String.format("[[\"getMessageList\", {\"filter\":{\"text\":\"tiramisu\"}}, \"#0\"]]"))
+            .when()
+            .post("/jmap")
+            .then()
+            .statusCode(200)
+            .body(ARGUMENTS + ".messageIds", contains(messageId));
+    }
+
+    @Test
+    public void getMessageListShouldListMessageThatHasBeenDeplacedInAMailboxWhereTheUserHasOnlyReadRight() throws Exception {
```

Comment:
This test worked before my bugfix because it does not trigger any real search on the content of the email.
If I modify it to add a filtering on the content then it will have fail before my bugfix like the one above